src/components/side-panel/add-elements-panel/add-elements-panel.tsx

```diff
@@ -51,11 +92,41 @@ export const AddElementsPanel = () => {
                         {renderElementButtons(category.elementTypes)}
                     </ElementButtonList>
                 </Category>
+            ));
+        } else {
+            return (
+                <NoResultsWrapper>
+                    <NotFoundIcon />
+                    <Text.Body>No results found.</Text.Body>
+                </NoResultsWrapper>
             );
-        });
+        }
     };
 
     return (
-        <Wrapper data-testid="add-elements-panel">{renderCategories()}</Wrapper>
+        <Wrapper data-testid="add-elements-panel">
+            <ElementSearch
+                placeholder="Search"
+                addon={{
+                    type: "custom",
+                    attributes: {
+                        children: (
+                            <MagnifierIcon
+                                onClick={handleMagnifierClick}
+                                data-testid="maginfying-glass"
+                            />
+                        ),
+                    },
+                    position: "left",
+                }}
+                allowClear={true}
+                value={searchInput}
+                onClear={handleOnClear}
+                onChange={(event) => generateSearchResults(event.target.value)}
+                ref={inputRef}
+            />
```

Comment:
instead of manually focusing on the search field with the `MagnifierIcon`, can we wrap it in a label instead?

```suggestion
            <label>
                <ElementSearch
                    placeholder="Search"
                    addon={{
                        type: "custom",
                        attributes: {
                            children: (
                                <MagnifierIcon
                                    onClick={handleMagnifierClick}
                                    data-testid="maginfying-glass"
                                />
                            ),
                        },
                        position: "left",
                    }}
                    allowClear={true}
                    value={searchInput}
                    onClear={handleOnClear}
                    onChange={(event) => generateSearchResults(event.target.value)}
                    ref={inputRef}
                />
            </label>
```

else you will need to use `IconWithButton` for the magnifier icon which comes with the proper accessibility tags